Source/Core/check.js

```diff
@@ -0,0 +1,111 @@
+/*global define*/
+define([
+    './defined',
+    './DeveloperError',
+    './isArray'
+], function(defined,
+            DeveloperError,
+            isArray) {
+    'use strict';
+
+    var exports = {
+        defined: checkDefined,
+        numeric: {
+            min: checkMin,
+            minMax: checkMinMax,
+            max: checkMax
+        },
+        type: {
+            array: checkArray,
+            boolean: checkBoolean,
+            function: checkFunction,
+            object: checkObject,
+            number: checkNumber,
+            string: checkString
+        }
+    };
+
+    var errors = {
+        defined: getUndefinedErrorMessage,
+        failedType: getFailedTypeErrorMessage
+    };
+
+    function getUndefinedErrorMessage(name) {
+        return name + ' is required but is undefined.';
+    }
+    function getFailedTypeErrorMessage(actual, expected) {
+        return 'Expected ' + expected + ', got ' + actual;
+    }
+
+    function checkMinMax(test, min, max) {
+        checkNumber(test);
+        checkNumber(max);
+        checkNumber(min);
+        if (min > max) {
+            throw new DeveloperError('Invalid condition: min (' + min + ') must be less than max (' + max + ')');
+        }
+        if (test > max || test < min) {
+            throw new DeveloperError('Invalid argument: expected ' + test + ' to be in range [' + min + ', ' + max + ']');
+        }
+    }
+
+    function checkMax(test, max) {
+        checkNumber(test);
+        checkNumber(max);
+        if (test > max) {
+            throw new DeveloperError('Expected ' + test + ' to be at most ' + max);
+        }
+    }
+
+    function checkMin(test, min) {
+        checkNumber(test);
+        checkNumber(min);
+        if (test < min) {
+            throw new DeveloperError('Expected ' + test + ' to be at least ' + min);
+        }
+    }
+
+    function checkDefined(test, name) {
+        if (test === undefined) {
```

Comment:
I was thinking perhaps I should avoid the extra function call for the sake of performance, though I should probably make it check for null as well as undefined in either case, for consistency with how parameter checks are currently handled.